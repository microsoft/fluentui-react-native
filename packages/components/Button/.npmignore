# Don't include test files in package
__tests__