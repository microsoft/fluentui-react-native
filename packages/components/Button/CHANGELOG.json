{
  "name": "@fluentui-react-native/button",
  "entries": [
    {
      "date": "Wed, 05 Aug 2020 22:50:11 GMT",
      "tag": "@fluentui-react-native/button_v0.10.0",
      "version": "0.10.0",
      "comments": {
        "none": [
          {
            "comment": "added descriptions",
            "author": "email not defined",
            "commit": "eb607f63531821f1b20b27081418045927b0f130",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Sat, 01 Aug 2020 00:34:34 GMT",
      "tag": "@fluentui-react-native/button_v0.10.0",
      "version": "0.10.0",
      "comments": {
        "minor": [
          {
            "comment": "specify types for tests",
            "author": "krsiler@microsoft.com",
            "commit": "f0021e6d95a1fa68c946a2dab2c797f32705fb9e",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Wed, 29 Jul 2020 22:38:20 GMT",
      "tag": "@fluentui-react-native/button_v0.9.27",
      "version": "0.9.27",
      "comments": {
        "none": [
          {
            "comment": "switch to using Lage for build sequencing",
            "author": "jasonmo@microsoft.com",
            "commit": "79677c8f3260dbf55e053db5983fb161c6fb44b5",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Tue, 28 Jul 2020 18:21:12 GMT",
      "tag": "@fluentui-react-native/button_v0.9.27",
      "version": "0.9.27",
      "comments": {
        "patch": [
          {
            "comment": "rename some token types and update Text",
            "author": "jasonmo@microsoft.com",
            "commit": "585007a447ba6dec0ae70be693879bbf7125c71b",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Thu, 23 Jul 2020 21:28:55 GMT",
      "tag": "@fluentui-react-native/button_v0.9.26",
      "version": "0.9.26",
      "comments": {
        "none": [
          {
            "comment": "Revert \"Change files\"",
            "author": "email not defined",
            "commit": "bac059cfd36f5a43185f3624e60bdda2d318c1b9",
            "package": "@fluentui-react-native/button"
          }
        ],
        "patch": [
          {
            "comment": "set backgroundColor on focus",
            "author": "lehon@microsoft.com",
            "commit": "481ee28dcee54178f442bc95bef174ca896a5951",
            "package": "@fluentui-react-native/button"
          },
          {
            "comment": "manually update versions",
            "author": "taamireh@microsoft.com",
            "commit": "72781079a1c236bdc2f8d9aa54bb2a904f8310f5",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Thu, 02 Jul 2020 21:06:06 GMT",
      "tag": "@fluentui-react-native/button_v0.9.22",
      "version": "0.9.22",
      "comments": {
        "none": [
          {
            "comment": "Revert \"Change files\"",
            "author": "email not defined",
            "commit": "bac059cfd36f5a43185f3624e60bdda2d318c1b9",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Wed, 17 Jun 2020 05:00:56 GMT",
      "tag": "@fluentui-react-native/button_v0.9.13",
      "version": "0.9.13",
      "comments": {
        "patch": [
          {
            "comment": "add memo-cache package and consume it in the core framework",
            "author": "jasonmo@microsoft.com",
            "commit": "27b5fc6790369ab629dcf7f93d6091d4913915fa",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Thu, 11 Jun 2020 23:53:15 GMT",
      "tag": "@fluentui-react-native/button_v0.9.0",
      "version": "0.9.0",
      "comments": {
        "minor": [
          {
            "comment": "add tooltip prop to Button",
            "author": "lehon@microsoft.com",
            "commit": "70f7d6d5972b384f2e8d4dd2626adda762c59206",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Tue, 09 Jun 2020 17:39:01 GMT",
      "tag": "@fluentui-react-native/button_v0.8.4",
      "version": "0.8.4",
      "comments": {
        "patch": [
          {
            "comment": "switch dependencies to use greater than semver until we reach 1.0.0",
            "author": "jasonmo@microsoft.com",
            "commit": "8461668e0387928c0831846b219188a6d4408295",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Sat, 06 Jun 2020 03:03:17 GMT",
      "tag": "@fluentui-react-native/button_v0.8.2",
      "version": "0.8.2",
      "comments": {
        "none": [
          {
            "comment": "support keyboardFocus for Link and add focus visuals",
            "author": "lehon@microsoft.com",
            "commit": "effb17ada2fd26c6ba8f7cbb4ab7d7ddf8d74c49",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Thu, 04 Jun 2020 23:23:26 GMT",
      "tag": "@fluentui-react-native/button_v0.8.0",
      "version": "0.8.0",
      "comments": {
        "minor": [
          {
            "comment": "Changing location of testID",
            "author": "safreibe@microsoft.com",
            "commit": "f46fc39f159f235f09ae5c3f8498d9ec36dac4fe",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Thu, 04 Jun 2020 22:15:30 GMT",
      "tag": "@fluentui-react-native/button_v0.7.1",
      "version": "0.7.1",
      "comments": {
        "patch": [
          {
            "comment": "Update snapshots to reflect fontSize in CSS pixels",
            "author": "krsiler@microsoft.com",
            "commit": "fb659219fcc84bb437515c97d816e82eb58e3fa9",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Mon, 01 Jun 2020 22:06:07 GMT",
      "tag": "@fluentui-react-native/button_v0.7.0",
      "version": "0.7.0",
      "comments": {
        "minor": [
          {
            "comment": "Support enter key for calling onClick on Button",
            "author": "lehon@microsoft.com",
            "commit": "399cead569e120926521d01c13a49088fe96036b",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Thu, 21 May 2020 20:00:42 GMT",
      "tag": "@fluentui-react-native/button_v0.6.0",
      "version": "0.6.0",
      "comments": {
        "minor": [
          {
            "comment": "add findNodeHandle in ButtonFocusTest",
            "author": "lehon@microsoft.com",
            "commit": "5ba3ac570ae17a0d3138302ad12807362dea06c1",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Wed, 20 May 2020 03:37:41 GMT",
      "tag": "@fluentui-react-native/button_v0.5.26",
      "version": "0.5.26",
      "comments": {
        "patch": [
          {
            "comment": "remove backgroundColor from focused _override setting to let hovered setting take precedence",
            "author": "lehon@microsoft.com",
            "commit": "4867e1dde6f8c5cdb3fe894e60f5bcbe8f3c2817",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Tue, 19 May 2020 23:37:15 GMT",
      "tag": "@fluentui-react-native/button_v0.5.25",
      "version": "0.5.25",
      "comments": {
        "patch": [
          {
            "comment": "Adding testID",
            "author": "safreibe@microsoft.com",
            "commit": "ab9641d24eba9201d41794fe303ef25155ef89e3",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Tue, 19 May 2020 20:06:57 GMT",
      "tag": "@fluentui-react-native/button_v0.5.23",
      "version": "0.5.23",
      "comments": {
        "patch": [
          {
            "comment": "ðŸ“¦ Update package descriptions",
            "author": "email not defined",
            "commit": "16687f10ee6e6115cb0756e936eecacb4e198b85",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Mon, 11 May 2020 20:15:56 GMT",
      "tag": "@fluentui-react-native/button_v0.5.15",
      "version": "0.5.15",
      "comments": {
        "patch": [
          {
            "comment": "bump react native win32 version to 0.62 and pull in matching tester",
            "author": "jasonmo@microsoft.com",
            "commit": "e0a1a7baa8be61b037ee65c49be22be60dc4b032",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Fri, 08 May 2020 18:49:53 GMT",
      "tag": "@fluentui-react-native/button_v0.5.14",
      "version": "0.5.14",
      "comments": {
        "patch": [
          {
            "comment": "change parameter format so include necessary props",
            "author": "lehon@microsoft.com",
            "commit": "ff5a5b0edb37bb35c9b2a4e817089a5a56551604",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Fri, 24 Apr 2020 19:41:08 GMT",
      "tag": "@fluentui-react-native/button_v0.5.4",
      "version": "0.5.4",
      "comments": {
        "patch": [
          {
            "comment": "more package version updates",
            "author": "jasonmo@microsoft.com",
            "commit": "1d2b333a0d29109c6734f43d949699ede91f3358",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Fri, 24 Apr 2020 19:11:27 GMT",
      "tag": "@fluentui-react-native/button_v0.5.3",
      "version": "0.5.3",
      "comments": {
        "patch": [
          {
            "comment": "renormalize files",
            "author": "jasonmo@microsoft.com",
            "commit": "dc062498d62d6f52c1d350d48cd463322f29471d",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Tue, 21 Apr 2020 17:03:45 GMT",
      "tag": "@fluentui-react-native/button_v0.5.0",
      "version": "0.5.0",
      "comments": {
        "minor": [
          {
            "comment": "Update button snapshots to reflect new fluent typeramp",
            "author": "krsiler@microsoft.com",
            "commit": "b034558f160d8815b11b16cd8d30e274e035c87e",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Tue, 21 Apr 2020 00:41:10 GMT",
      "tag": "@fluentui-react-native/button_v0.4.1",
      "version": "0.4.1",
      "comments": {
        "patch": [
          {
            "comment": "update beachball hook to use new prepublish strategy",
            "author": "jasonmo@microsoft.com",
            "commit": "7bc030df722a6fcb6f693658b9d80f4617a26713",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Fri, 17 Apr 2020 22:36:03 GMT",
      "tag": "@fluentui-react-native/button_v0.4.0",
      "version": "0.4.0",
      "comments": {
        "minor": [
          {
            "comment": "yarn.lock",
            "author": "warleu@microsoft.com",
            "commit": "205e7b7e88151a17f2560ecdaec1c6e13c5825b7",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Fri, 17 Apr 2020 16:48:18 GMT",
      "tag": "@fluentui-react-native/button_v0.3.8",
      "version": "0.3.8",
      "comments": {
        "patch": [
          {
            "comment": "Revert \"Merge pull request #136 from ksiler/fluent-variant-support\"",
            "author": "krsiler@microsoft.com",
            "commit": "302097d2dc2dc50680be2ff3747b9c244501d7d5",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Thu, 16 Apr 2020 23:57:38 GMT",
      "tag": "@fluentui-react-native/button_v0.3.7",
      "version": "0.3.7",
      "comments": {
        "patch": [
          {
            "comment": "publish with correct main/module references",
            "author": "jasonmo@microsoft.com",
            "commit": "6709e9eed6e08029114e8d893d27d91b99f73d0a",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Mon, 13 Apr 2020 22:48:23 GMT",
      "tag": "@fluentui-react-native/button_v0.3.6",
      "version": "0.3.6",
      "comments": {
        "patch": [
          {
            "comment": "Update button snapshot",
            "author": "krsiler@microsoft.com",
            "commit": "232a97794dc044cec9671e5cca4cf204d881f614",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Thu, 09 Apr 2020 21:14:58 GMT",
      "tag": "@fluentui-react-native/button_v0.3.5",
      "version": "0.3.5",
      "comments": {
        "patch": [
          {
            "comment": "enable button snapshot test and update merge to fix bug",
            "author": "jasonmo@microsoft.com",
            "commit": "232a97794dc044cec9671e5cca4cf204d881f614",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Thu, 09 Apr 2020 18:39:15 GMT",
      "tag": "@fluentui-react-native/button_v0.3.3",
      "version": "0.3.3",
      "comments": {
        "patch": [
          {
            "comment": "merge conflicts",
            "author": "ppatboyd@outlook.com",
            "commit": "232a97794dc044cec9671e5cca4cf204d881f614",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Fri, 03 Apr 2020 20:40:51 GMT",
      "tag": "@fluentui-react-native/button_v0.3.1",
      "version": "0.3.1",
      "comments": {
        "patch": [
          {
            "comment": "switch the bin name from just-script to fluent-scripts to disambiguate names",
            "author": "jasonmo@microsoft.com",
            "commit": "232a97794dc044cec9671e5cca4cf204d881f614",
            "package": "@fluentui-react-native/button"
          }
        ]
      }
    },
    {
      "date": "Tue, 31 Mar 2020 18:01:09 GMT",
      "tag": "@fluentui-react-native/button_v0.3.0",
      "version": "0.3.0",
      "comments": {
        "minor": [
          {
            "comment": "Checkbox Final V1 Changes",
            "author": "safreibe@microsoft.com",
            "commit": "0a82a531a8dd9cd29d1799badaa49733ef7c8ccf"
          }
        ]
      }
    },
    {
      "date": "Mon, 30 Mar 2020 16:27:00 GMT",
      "tag": "@fluentui-react-native/button_v0.2.7",
      "version": "0.2.7",
      "comments": {
        "patch": [
          {
            "comment": "Update main attribute to specify TypeScript index file.",
            "author": "kinhln@microsoft.com",
            "commit": "ab291fb90d584c81310aa6559d05a360e86f97de"
          }
        ]
      }
    },
    {
      "date": "Fri, 27 Mar 2020 04:10:56 GMT",
      "tag": "@fluentui-react-native/button_v0.2.6",
      "version": "0.2.6",
      "comments": {
        "none": [
          {
            "comment": "shared package configs",
            "author": "ppatboyd@outlook.com",
            "commit": "1471793de9a859b9908ef50069aeb673d06e5b42"
          }
        ]
      }
    },
    {
      "date": "Wed, 18 Mar 2020 00:37:12 GMT",
      "tag": "@fluentui-react-native/button_v0.2.3",
      "version": "0.2.3",
      "comments": {
        "none": [
          {
            "comment": "no change",
            "author": "email not defined",
            "commit": "eb75093eefe61d9235cb01a8adcafd105e57351c"
          }
        ]
      }
    },
    {
      "date": "Mon, 16 Mar 2020 23:03:24 GMT",
      "tag": "@fluentui-react-native/button_v0.2.0",
      "version": "0.2.0",
      "comments": {
        "minor": [
          {
            "comment": "turn on publishing",
            "author": "jasonmo@microsoft.com",
            "commit": "86f6fc7932479f6afcb8f77c539dd4e0d4e436d7"
          }
        ]
      }
    }
  ]
}
