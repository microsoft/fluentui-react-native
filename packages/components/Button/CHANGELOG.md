# Change Log - @fluentui-react-native/button

This log was last generated on Thu, 09 Apr 2020 21:14:58 GMT and should not be manually modified.

## 0.3.5
Thu, 09 Apr 2020 21:14:58 GMT

### Patches

- enable button snapshot test and update merge to fix bug (jasonmo@microsoft.com)
## 0.3.3
Thu, 09 Apr 2020 18:39:15 GMT

### Patches

- merge conflicts (ppatboyd@outlook.com)
## 0.3.1
Fri, 03 Apr 2020 20:40:51 GMT

### Patches

- switch the bin name from just-script to fluent-scripts to disambiguate names (jasonmo@microsoft.com)
## 0.3.0
Tue, 31 Mar 2020 18:01:09 GMT

### Minor changes

- Checkbox Final V1 Changes (safreibe@microsoft.com)
## 0.2.7
Mon, 30 Mar 2020 16:27:00 GMT

### Patches

- Update main attribute to specify TypeScript index file. (kinhln@microsoft.com)
## 0.2.0
Mon, 16 Mar 2020 23:03:24 GMT

### Minor changes

- turn on publishing (jasonmo@microsoft.com)