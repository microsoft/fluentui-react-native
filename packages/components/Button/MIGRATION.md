# Button Migration

## Migration from v0 Button

In the short term, the new `Button` control is named `ButtonV1` while it clashes with the existing older control. Once we deprecate the old control, it will be renamed to `Button`. It may be useful to rename the control to `Button` using the import syntax to simplify the rename:

```ts
import { ButtonV1 as Button } from '@fluentui-react-native/button'
```

### Component renames

Primary and Stealth buttons now map to `Button`:

| v0 `Button`                             | v1 `Button`                                          |
| --------------------------------------- | ---------------------------------------------------- |
| `<Button content="Hello, world" />`        | `<Button>Hello, world</Button>`                      |
| `<PrimaryButton content="Hello, world" />` | `<Button appearance='primary'>Hello, world</Button>` |
| `<StealthButton content="Hello, world" />` | `<Button appearance='subtle'>Hello, world</Button>`  |

### Props that remain as is

- Any props that are part of `ViewProps`
- `children`
- `componentRef`
- `icon`
- `onClick`
- `testID`
- `tooltip`

### Props no longer supported with an equivalent functionality in v1 Button

- `content` => Pass the content as `children` instead
- `endIcon` => Use `after` value for `iconPosition` prop and pass icon information into `icon` prop instead
- `startIcon` => Use `before` value for `iconPosition` prop and pass icon information into `icon` prop instead

### Props no longer supported without an equivalent functionality in v1 Button

- Cannot use both `startIcon` and `endIcon` at the same time in v1

### Tokens that remain as is

- Any props that are part of `FontTokens`, `IForegroundColorTokens`, `IBackgroundColorTokens`, `IBorderTokens`
- `iconColor`
- `iconSize`
- `iconWeight`

### Tokens no longer supported with an equivalent functionality in v1 Button

- `contentPadding` => Set on `style` property of `root` property under `slotProps` in `compose` API
- `iconColorHovered` => Use `iconColor` token under the `hovered` state token
- `iconColorPressed` => Use `iconColor` token under the `pressed` state token

### Tokens no longer supported without an equivalent functionality in v1 Button

- `contentPaddingFocused` => Removed in favor of using native based focus visuals
- `wrapperBorderColor` => Removed in favor of using native based focus visuals

### Slots no longer supported with an equivalent functionality in v1 Button

- `borderWrapper` has been removed in favor of using native focus visuals. Use `root` instead.
- `stack` has been removed. Use `root` instead.

### Updating ThemeProvider

If you are using the older theme provider `ThemeProvider` from `@uifabricshared/theming-react-native`, you will need to update the `ThemeProvider` to pull from `@fluentui-react-native/theme` to have the control work properly with themes. Please see [this page](../../../docs/pages/Guides/UpdateThemeProvider.md) for guidance.

### Migrating customized Buttons

Please see [this page](../../../docs/pages/Guides/UpdatingCustomize.md) for guidance on how to move from the old `customize` API to the new one.

If you were using `PrimaryButton` or `StealthButton`, any color customizations will need to be applied under the `primary` and `subtle` tokens, respectively:

```ts
const Custom = StealthButton.customize({
  tokens: {
    borderWidth: 0,
    color: 'white',
    backgroundColor: 'red',
  }
});
```

would become

```ts
const Custom = Button.customize({
  subtle: {
    borderWidth: 0,
    color: 'white',
    backgroundColor: 'red',
  }
});
```

### Migrating composed Buttons

Please see [this page](../../../docs/pages/Guides/UpdatingCustomize.md) for guidance on how to move from the old `customize` API to the new one.


## Porting from FluentUI v9 Button

The FURN button cannot be used in place of a FluentUI button - these buttons are intended to be used on their respective platforms. See [this porting guide](../../../docs/pages/Guides/PortingFromFluentUI.md) for general guidance on coming from FluentUI to FURN.

### Props that remain as is

- `children`
- `disabled`
- `icon`
- `onClick`
- `testID`
- `tooltip`

### Props renamed to align with ReactNative

No `Button` specific renames.

### Prop differences due to technical differences and limitations

- `icon` takes in a props object instead of the JSX element itself. This is due to framework differences from FluentUI.
- `iconOnly` must be supplied for components do not have any text content for them to be styled correctly. This is due to framework differences from FluentUI.
- `ref` is exposed as `componentRef`, similar to previous versions of FluentUI. This is due to framework differences from FluentUI.

### Other Prop differences

- `appearance` does not include values for `transparent` or `outline` as neither currently are used by native platforms.
- `iconPosition` uses `before` and `after` as values instead of `left` and `right` for better right to left support.

## Property mapping

| v0 `Button`    | v1 `Button`    |
| -------------- | -------------- |
| `componentRef` | `componentRef` |
| `content`      |                |
| `disabled`     | `disabled`     |
| `endIcon`      | `iconPosition` |
| `icon`         | `icon`         |
| `onClick`      | `onClick`      |
| `statIcon`     | `iconPosition` |
| `testId`       | `testId`       |
| `tooltip`      | `tooltip`      |
