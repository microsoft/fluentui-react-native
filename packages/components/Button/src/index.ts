export { buttonName } from './deprecated/Button.types';
export type {
  IButtonInfo,
  IButtonProps,
  IButtonRenderData,
  IButtonSlotProps,
  IButtonState,
  IButtonTokens,
  IButtonType,
} from './deprecated/Button.types';
export { Button } from './deprecated/Button';
export { PrimaryButton } from './deprecated/PrimaryButton';
export { StealthButton } from './deprecated/StealthButton';
export type {
  ButtonSize,
  ButtonAppearance,
  ButtonShape,
  ButtonCoreTokens,
  ButtonTokens,
  ButtonCoreProps,
  ButtonProps,
  ButtonInfo,
  ButtonSlotProps,
  ButtonType,
} from './Button.types';
export { buttonName as buttonNameV1 } from './Button.types';
export { Button as ButtonV1 } from './Button';
export { useButton } from './useButton';
export { CompoundButton, compoundButtonName } from './CompoundButton';
export type { CompoundButtonProps, CompoundButtonSlotProps, CompoundButtonTokens, CompoundButtonType } from './CompoundButton';
export { FAB, fabName } from './FAB';
export type { FABSize, FABSlotProps, FABProps, FABTokens, FABType } from './FAB';
export { ToggleButton, toggleButtonName, useToggleButton } from './ToggleButton';
export type { ToggleButtonProps, ToggleButtonSlotProps, ToggleButtonInfo, ToggleButtonTokens, ToggleButtonType } from './ToggleButton';
