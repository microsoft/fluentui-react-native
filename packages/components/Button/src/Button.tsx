/** @jsx withSlots */
import * as React from 'react';
import { Image, View } from 'react-native';
import { IButtonSlotProps, IButtonState, IButtonProps, IButtonRenderData, buttonName, IButtonType } from './Button.types';
import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
import { Text } from '@fluentui-react-native/text';
import { settings } from './Button.settings';
import { backgroundColorTokens, borderTokens, textTokens, foregroundColorTokens, getPaletteFromTheme } from '@fluentui-react-native/tokens';
import { filterViewProps, filterImageProps } from '@fluentui-react-native/adapters';
import { mergeSettings } from '@uifabricshared/foundation-settings';
import { useAsPressable, useKeyCallback, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';

export const Button = compose<IButtonType>({
  displayName: buttonName,
  usePrepareProps: (userProps: IButtonProps, useStyling: IUseComposeStyling<IButtonType>) => {
    const {
      icon,
      content,
      onAccessibilityTap = userProps.onClick,
      accessibilityLabel = userProps.content,
      testID,
      onClick,
      ...rest
    } = userProps;
    // attach the pressable state handlers
    const pressable = useAsPressable({ ...rest, onPress: onClick });
    const onKeyUp = useKeyCallback(onClick, ' ', 'Enter');
    // set up state
    const state: IButtonState = {
      info: {
        ...pressable.state,
        disabled: !!userProps.disabled,
        content: !!content,
        icon: !!icon,
      },
    };

    const buttonRef = useViewCommandFocus(userProps.componentRef);
    // grab the styling information, referencing the state as well as the props
    const styleProps = useStyling(userProps, (override: string) => state.info[override] || userProps[override]);
    // create the merged slot props

    const slotProps = mergeSettings<IButtonSlotProps>(styleProps, {
      root: {
        ...pressable.props,
        ref: buttonRef,
        onAccessibilityTap: onAccessibilityTap,
        accessibilityLabel: accessibilityLabel,
        accessibilityState: { disabled: state.info.disabled },
        onKeyUp: onKeyUp,
      },
      content: { children: content, testID: testID },
      icon: { source: icon },
    });

    return { slotProps, state };
  },
  settings,
  render: (Slots: ISlots<IButtonSlotProps>, renderData: IButtonRenderData, ...children: React.ReactNode[]) => {
    const info = renderData.state!.info;

    // We shouldn't have to specify the source prop on Slots.icon, here, but we need another drop from @uifabricshared
    return (
      <Slots.root>
        <Slots.stack>
          {info.icon && <Slots.icon source={renderData.slotProps!.icon.source} />}
          {info.content && <Slots.content />}
          {children}
        </Slots.stack>
      </Slots.root>
    );
  },
  slots: {
    root: View,
    stack: { slotType: View, filter: filterViewProps },
    icon: { slotType: Image as React.ComponentType<object>, filter: filterImageProps },
    content: Text,
  },
  styles: {
    root: [backgroundColorTokens, borderTokens],
    stack: [],
    icon: [{ source: 'iconColor', lookup: getPaletteFromTheme, target: 'tintColor' }],
    content: [textTokens, foregroundColorTokens],
  },
});

export default Button;
