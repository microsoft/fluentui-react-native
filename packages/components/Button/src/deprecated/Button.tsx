/** @jsxRuntime classic */
/** @jsx withSlots */
import * as React from 'react';
import { View } from 'react-native';

import { filterViewProps } from '@fluentui-react-native/adapters';
import { Icon, createIconProps } from '@fluentui-react-native/icon';
import { useAsPressable, useKeyProps, useViewCommandFocus, useOnPressWithFocus } from '@fluentui-react-native/interactive-hooks';
import { Text } from '@fluentui-react-native/text';
import { backgroundColorTokens, borderTokens, textTokens, foregroundColorTokens, getPaletteFromTheme } from '@fluentui-react-native/tokens';
import type { ISlots } from '@uifabricshared/foundation-composable';
import { withSlots } from '@uifabricshared/foundation-composable';
import type { IUseComposeStyling } from '@uifabricshared/foundation-compose';
import { compose } from '@uifabricshared/foundation-compose';
import { mergeSettings } from '@uifabricshared/foundation-settings';

import { settings } from './Button.settings';
import type { IButtonSlotProps, IButtonState, IButtonProps, IButtonRenderData, IButtonType } from './Button.types';
import { buttonName } from './Button.types';

/**
 * @deprecated This component is deprecated in favor of ButtonV1. This Button will be removed when the package moves to 1.0.0.
 * At that point, ButtonV1 will be renamed to Button. Please see MIGRATION.md for details on how to move to the new Button.
 */
export const Button = compose<IButtonType>({
  displayName: buttonName,
  usePrepareProps: (userProps: IButtonProps, useStyling: IUseComposeStyling<IButtonType>) => {
    const defaultComponentRef = React.useRef(null);
    const {
      icon,
      startIcon,
      endIcon,
      content,
      onAccessibilityTap = userProps.onClick,
      accessibilityLabel = userProps.content,
      componentRef = defaultComponentRef,
      testID,
      onClick,
      ...rest
    } = userProps;

    // Ensure focus is placed on button after click
    const onPressWithFocus = useOnPressWithFocus(componentRef, onClick);
    // attach the pressable state handlers
    const pressable = useAsPressable({ ...rest, onPress: onPressWithFocus });
    const onKeyProps = useKeyProps(onClick, ' ', 'Enter');
    // set up state
    const state: IButtonState = {
      info: {
        ...pressable.state,
        disabled: !!userProps.disabled,
        content: !!content,
        startIcon: !!startIcon || !!icon,
        endIcon: !!endIcon,
      },
    };

    const buttonRef = useViewCommandFocus(componentRef);
    // grab the styling information, referencing the state as well as the props
    const styleProps = useStyling(userProps, (override: string) => state.info[override] || userProps[override]);
    // create the merged slot props

    const slotProps = mergeSettings<IButtonSlotProps>(styleProps, {
      root: {
        ...pressable.props,
        ref: buttonRef,
        onAccessibilityTap: onAccessibilityTap,
        accessibilityLabel: accessibilityLabel,
        accessibilityState: { disabled: state.info.disabled },
        ...onKeyProps,
        testID,
      },
      content: { children: content },
      startIcon: createIconProps(startIcon || icon),
      endIcon: createIconProps(endIcon),
    });

    return { slotProps, state };
  },
  settings,
  render: (Slots: ISlots<IButtonSlotProps>, renderData: IButtonRenderData, ...children: React.ReactNode[]) => {
    const info = renderData.state!.info;

    return (
      <Slots.root>
        <Slots.borderWrapper>
          <Slots.stack>
            {info.startIcon && <Slots.startIcon />}
            {info.content && <Slots.content />}
            {children}
            {info.endIcon && <Slots.endIcon />}
          </Slots.stack>
        </Slots.borderWrapper>
      </Slots.root>
    );
  },
  slots: {
    root: View,
    stack: { slotType: View, filter: filterViewProps },
    borderWrapper: { slotType: View, filter: filterViewProps },
    startIcon: { slotType: Icon as React.ComponentType },
    content: Text,
    endIcon: { slotType: Icon as React.ComponentType },
  },
  styles: {
    root: [backgroundColorTokens, borderTokens],
    stack: [],
    borderWrapper: [{ source: 'wrapperBorderColor', lookup: getPaletteFromTheme, target: 'borderColor' }],
    startIcon: [{ source: 'iconColor', lookup: getPaletteFromTheme, target: 'color' }],
    content: [textTokens, foregroundColorTokens],
    endIcon: [{ source: 'iconColor', lookup: getPaletteFromTheme, target: 'color' }],
  },
});

export default Button;
