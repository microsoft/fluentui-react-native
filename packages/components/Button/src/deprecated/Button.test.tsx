import * as React from 'react';

import * as renderer from 'react-test-renderer';

import { Button } from './Button';

it('Button default', () => {
  const tree = renderer.create(<Button content="Default Button" />).toJSON();
  expect(tree).toMatchSnapshot();
});
