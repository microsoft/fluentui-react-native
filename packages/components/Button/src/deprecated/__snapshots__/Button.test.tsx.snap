// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button default 1`] = `
<View
  accessibilityLabel="Default Button"
  accessibilityRole="button"
  accessibilityState={
    {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "flex-start",
      "alignSelf": "flex-start",
      "backgroundColor": "#f3f2f1",
      "borderColor": "#8a8886",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 24,
          "minWidth": 32,
          "paddingEnd": 6,
          "paddingStart": 6,
        }
      }
    >
      <Text
        style={
          {
            "color": "#323130",
            "fontFamily": "System",
            "fontSize": 13,
            "fontWeight": "400",
            "margin": 0,
            "marginEnd": 2,
            "marginStart": 2,
          }
        }
      >
        Default Button
      </Text>
    </View>
  </View>
</View>
`;
